src/components/leaderboards/Leaderboard/index.js

```diff
@@ -51,95 +51,24 @@ module.exports = React.createClass({
   },
 
-  componentWillMount: function() {
-    this.loadLeaderboard();
+  componentDidMount: function() {
+    addEventListener(window, 'resize', this.setChildWidth);
   },
 
-  getEndpoint: function() {
-    var endpoint;
-
-    var props = this.props;
-    if (props.country) {
-      if (props.campaignSlug) { endpoint = campaigns.leaderboardBySlug.bind(campaigns, props.country, props.campaignSlug); }
-      if (props.charitySlug)  { endpoint = charities.leaderboardBySlug.bind(charities, props.country, props.charitySlug); }
-    } else {
-      if (props.campaignUid)  { endpoint = campaigns.leaderboard.bind(campaigns, props.campaignUid); }
-      if (props.charityUid)   { endpoint = charities.leaderboard.bind(charities, props.charityUid); }
-    }
-
-    if (!endpoint && console && console.log) {
-      console.log("Leaderboard widget requires 'campaignUid' or 'charityUid'. If 'country' is given then 'campaignSlug' or 'charitySlug' may be used instead. ");
-    }
-
-    return (endpoint || function(type, limit, callback) { callback(null); });
-  },
-
-  loadLeaderboard: function() {
-    this.setState({
-      isLoading: true
-    });
-
-    var endpoint = this.getEndpoint();
-    endpoint(this.props.type, this.props.limit, this.processLeaderboard, { includePages: true });
+  componentWillUnmount: function() {
+    removeEventListener(window, 'resize', this.setChildWidth);
   },
 
-  processLeaderboard: function(result) {
-    var pages = result && result.leaderboard && result.leaderboard.pages ? result.leaderboard.pages : [];
-    var leaderboard = _.map(pages, this.processPage);
-
-    this.rankLeaderboard(leaderboard);
-
+  setChildWidth: _.debounce(function() {
     this.setState({
-      isLoading: false,
-      boardData: this.paginateLeaderboard(leaderboard)
+      childWidth: this.getChildrenWidth(this.props.childWidth, this.props.pageSize)
     });
-  },
-
-  processPage: function(page) {
-    return {
-      id: page.id,
-      name: page.name,
-      url: page.url,
-      isoCode: page.amount.currency.iso_code,
-      amount:  page.amount.cents,
-      totalMembers: page.team_member_uids.length,
-      imgSrc: page.image.large_image_url,
-      medImgSrc: page.image.medium_image_url
-    };
-  },
-
-  rankLeaderboard: function(leaderboard) {
-    var rank = 1;
-    var prevItem = null;
-
-    _.forEach(leaderboard, function(item, i) {
-      if (prevItem && item.amount != prevItem.amount) {
-        rank = i + 1;
-      }
-
-      item.rank = rank;
-      prevItem = item;
-    });
-  },
-
-  paginateLeaderboard: function(leaderboard) {
-    var pageSize         = this.props.pageSize;
-    var pagedLeaderboard = [];
-
-    for (var i = 0; i < leaderboard.length; i += pageSize) {
-      pagedLeaderboard.push(leaderboard.slice(i,i + pageSize));
-    }
-
-    return pagedLeaderboard;
-  },
-
-  formatAmount: function(amount) {
-    return this.t('symbol') + numeral(amount / 100).format(this.props.currencyFormat);
-  },
+  }, 100),
```

Comment:
}, 100, { trailing: true }),

to fire it when the timer ends, not when it starts. Otherwise it will set the width as soon as they start resizing the page (old width), not when they finish (new width).